### PR TITLE
Feature/cplat add ip address warning

### PR DESCRIPTION
**Pull request by pselugar**
## Description

* Added new Parameter `-PublicIpSku` to the `NewAzVM` cmdlet with acceptable values : "Basic" and "Standard". 
* Added Generic Breaking Change PublicIpSku Warning and Overridden `-Zone` logic when `-PublicIpSku` is explicitly provided.
*  Work Item for ref : [PublicIpSku Warning](https://msazure.visualstudio.com/One/_workitems/edit/14469917)
* Debug and tested the changes.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -20,7 +20,8 @@
 
 -->
 ## Upcoming Release
-
+* Added new Parameter `-PublicIpSku` to the `NewAzVM` cmdlet with acceptable values : "Basic" and "Standard". 
+* Added Generic Breaking Change PublicIpSku Warning and Overridden `-Zone` logic when `-PublicIpSku` is explicitly provided.
 ## Version 4.29.0
```
Comment (grizzlytheodore) [inline src/Compute/Compute/ChangeLog.md:25]: empty line here

Comment (pselugar) [reply src/Compute/Compute/ChangeLog.md:25]: Good catch, will work on that

File: src/Compute/Compute/Compute.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
 
   <PropertyGroup>
     <ModuleAlcEntryAssembly>Microsoft.Azure.Commands.Compute.AlcWrapper</ModuleAlcEntryAssembly>
+    <OutputType>Library</OutputType>
```
Comment (vidai-msft) [inline src/Compute/Compute/Compute.csproj:30]: Is it necessary to make this line of change?

Comment (pselugar) [reply src/Compute/Compute/Compute.csproj]: Not necessarily, might have been added as part of SDK version change. I removed it in the next revision.

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 3]
```diff
@@ -50,10 +50,11 @@
 using SM = Microsoft.Azure.PowerShell.Cmdlets.Compute.Helpers.Storage.Models;
 using Microsoft.Azure.Commands.Compute;
 using Microsoft.Azure.PowerShell.Cmdlets.Compute.Helpers.Network.Models;
-
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
 
 namespace Microsoft.Azure.Commands.Compute
 {
+    [GenericBreakingChange("It is recommended to use parameter \"-PublicIpSku Standard\" in order to create a new VM with a Standard public IP.Specifying zone(s) using the \"-Zone\" parameter will also result in a Standard public IP.If \"-Zone\" is not specified, the VM will be created with a Basic public IP instead.Please note that the Standard SKU IPs will become the default behavior for VM creation in the future")]
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:57]: Spaces after periods

Comment (grizzlytheodore) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:57]: currently: If \"-Zone\" is not specified, the VM will be created with a Basic public IP instead.
I think we should specify:
"If zone AND PublicIPSku are not specificed,"

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 4]
```diff
@@ -510,6 +522,19 @@ public async Task<ResourceConfig<VirtualMachine>> CreateConfigAsync()
                     }
                 }
 
+                //Override Zone logic if PublicIpSku is explicitly provided
+                PublicIPAddressStrategy.Sku publicIpSku;
+                if (_cmdlet.PublicIpSku != null) {
+                    if (_cmdlet.PublicIpSku != "Basic" || _cmdlet.PublicIpSku != "Standard")
+                    {
+                        throw new InvalidDataException("Invalid data entry. Acceptable values for PublicIpSku are \"Basic\" or \"Standard\" only");
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:530]: Lets either do "-PublicIpSku" or parameter PublicIpSku

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 5]
```diff
@@ -510,6 +522,19 @@ public async Task<ResourceConfig<VirtualMachine>> CreateConfigAsync()
                     }
                 }
 
+                //Override Zone logic if PublicIpSku is explicitly provided
+                PublicIPAddressStrategy.Sku publicIpSku;
+                if (_cmdlet.PublicIpSku != null) {
+                    if (_cmdlet.PublicIpSku != "Basic" || _cmdlet.PublicIpSku != "Standard")
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:528]: should it be a AND logic instead?

File: src/Compute/Compute/help/New-AzVM.md [hunk 6]
```diff
@@ -894,7 +912,6 @@ The name of a new (or existing) network security group (NSG) for the created VM
 Type: System.String
 Parameter Sets: SimpleParameterSet, DiskFileParameterSet
 Aliases:
-
```
Comment (grizzlytheodore) [inline src/Compute/Compute/help/New-AzVM.md:897]: lets put this space back

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:57]: I copied this warning (provided by Annie) from the Work Item. Should I still modify it or keep it as it was given by Annie?

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 5 above)
Comment (pselugar) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:528]: OR works too but AND logic makes more sense and sounds correct. Will make the change

File: src/Compute/Compute/help/New-AzVM.md (hunk 6 above)
Comment (pselugar) [reply src/Compute/Compute/help/New-AzVM.md:897]: Done

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:57]: Never mind, "If zone AND PublicIPSku are not specificed," that part makes the warning sounds clearer to users. Good catch!

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 5 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs]: I do not think OR would work. OR statement make this if statement satisfy no matter what

**Review by vidai-msft — APPROVED** [review]
Looks good!

**Pull request merged**